### DISCUSSION TITLE
Module build failed when re-exporting types

### DISCUSSION THREAD

**Discussion by boy51** [nrwl/nx#9012] [General]
Hi, I'm pretty new to nx and just trying to get the hang of it.

I'm migrating a project to nx. Currently I'm trying to get a react app running which depends on a library that exports interfaces.

When building react-app, I get the error

```ERROR in ../../libs/api-client/src/index.ts
Module build failed (from ../../node_modules/@nrwl/web/src/utils/web-babel-loader.js):     
SyntaxError: ...\libs\api-client\src\index.ts: Support for the experimental syntax 'flow' isn't currently enabled (7:8):

   5 | export { CancelablePromise, CancelError } from './core/CancelablePromise';
   6 | export { OpenAPI } from './core/OpenAPI';
>  7 | export type { OpenAPIConfig } from './core/OpenAPI';

Add @babel/preset-flow (https://git.io/JfeDn) to the 'presets' section of your Babel config to enable transformation.
If you want to leave it as-is, add @babel/plugin-syntax-flow (https://git.io/vb4yb) to the 
'plugins' section to enable parsing.
```

Clearly babel is getting stuck when consuming a file that is re-exporting a type.

What's confusing me is that I don't use flow, so I don't think adding `@babel/plugin-syntax-flow` as suggested in the error is the way to go.

So I see 2 options to fix:
1. Run library through tsc
2. Fix babel config of react app

I tried to go with number 1, trying to build the library before the app is built. When I add the build step to the library and run app:build, it runs library:build first as expected, but when building the app the error still comes up.

I expected babel to use the already-built output from dist, but its still referencing the file in source code and giving the same error.

Does someone have a suggestion to fix? I know this is more to do with babel but I'm sure there is a fix that I'm unaware.

**Comment by tbinna** [comment]
A solution for this is documented here: https://github.com/nrwl/nx/discussions/8600
